appservice/src/IDeploymentsClient.ts

```diff
@@ -3,14 +3,19 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { SiteConfigResource, SiteSourceControl } from 'azure-arm-website/lib/models';
+import { SiteConfigResource, SiteSourceControl, StringDictionary, User } from 'azure-arm-website/lib/models';
 import { KuduClient } from 'vscode-azurekudu';
 
 export interface IDeploymentsClient {
 
     fullName: string;
     isFunctionApp: boolean;
+    isLinux: boolean;
+    gitUrl: string | undefined;
+    listApplicationSettings(): Promise<StringDictionary>;
```

Comment:
This doesn't feel right.  ApplicationSettings shouldn't be in the DeploymentsClient, I'd think.

Reply:
Yeah it didn't feel right to me either. However `verifyNoRunFromPackageSetting` uses the client from `localGitDeploy`... I could only call verifyNoRunPackageSetting if `client instanceof IAppSettingsTreeItem` maybe.